homeassistant/components/huawei_solar/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for Huawei Solar integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huawei_solar import (
+    AsyncHuaweiSolar,
+    ConnectionException,
+    ReadException,
+    register_names as rn,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_SLAVE_IDS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 502
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Required(CONF_SLAVE_IDS, default="0"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    inverter = None
+    try:
+        inverter = await AsyncHuaweiSolar.create(
+            host=data[CONF_HOST],
+            port=data[CONF_PORT],
+            slave=data[CONF_SLAVE_IDS][0],
+        )
+
+        model_name, serial_number = await inverter.get_multiple(
+            [rn.MODEL_NAME, rn.SERIAL_NUMBER]
+        )
+
+        _LOGGER.info(
+            f"Successfully connected to inverter {model_name.value} with SN {serial_number.value}"
+        )
+
+        # Also validate the other slave-ids
+        for slave_id in data[CONF_SLAVE_IDS][1:]:
+            try:
+                slave_model_name, slave_serial_number = await inverter.get_multiple(
+                    [rn.MODEL_NAME, rn.SERIAL_NUMBER], slave_id
+                )
+
+                _LOGGER.info(
+                    f"Successfully connected to slave inverter {slave_id}: {slave_model_name.value} with SN {slave_serial_number.value}"
+                )
+            except Exception as err:  # pylint: disable=broad-except
```

Comment:
Are there no specific exceptions in the module that you can catch? Broad exceptions like this are not really nice.